### DISCUSSION TITLE
[Bug]: Tweak doesn't load when using 3rd party swift framework

### DISCUSSION THREAD

**Discussion by gitusehub** [theos/theos#672] [Questions]
### What are the steps to reproduce this issue?

1. Make project
2. Compile a swift project into a framework (Example: Alamofire)
3. Add the .framework to theos /lib/ and/or (/vendor/lib) folder
4. add "TWEAK_NAME_EXTRA_FRAMEWORKS = Alamofire" in Makefile
5. run "make package install"
6. No error in the compilation process
7. After respring, nothing happens, the tweak successfully installed, but when checking with FLEX, the tweak is not loaded into the process, but when removing the "TWEAK_NAME_EXTRA_FRAMEWORKS = Alamofire", and compile it again, the tweak works just fine, the problem only occurs when adding the framework into the project

### What happens?

After respring, nothing happens, the tweak successfully installed, but when checking with FLEX, the tweak is not loaded into the process, but when removing the "TWEAK_NAME_EXTRA_FRAMEWORKS = Alamofire", and compile it again, the tweak works just fine, the problem only occurs when adding the framework into the project

### What were you expecting to happen?

Expecting the tweak to load into the process, but no success, there is no error at all, during the compilation, or during the installation

### Paste any relevant logs, error output, etc.

```shell
No error log at all
```

### What OS are you seeing the problem on?

iOS

### What OS version does it have installed?

macOS 12.6

### What toolchain and version are you using?

xcode 14

### Which SDK version are you using?

iOS 13.0 & iOS 14.5

### What OS is your client device running?

iOS

### What OS version does it have installed?

iOS 14.x

### Which device model is it?

iPhone 8

### Which jailbreak is it using (if relevant)?

Unc0ver

**Comment by kirb** [comment]
Did you install Alamofire.framework onto the device? Theos intentionally doesn't embed frameworks for you, because there are a number of ways you can do so, and also because it's not super recommended for tweaks to be injecting common libraries like Alamofire that might conflict with the copy used by an app or another tweak.

If you run `otool -L Alamofire.framework/Alamofire` you'll see the install name on the first line. It may be something like `@rpath/Frameworks/Alamofire.framework/Alamofire`, in which case you can either place it at that path (less the `@rpath`) relative to where your own binary is in the filesystem, or tell the linker to add another rpath search path wherever you'd like to put it:

```make
MyTweak_LDFLAGS += -rpath '/Library/Application Support/MyTweak'
```

You can also confirm what exactly is going on by filtering for "dyld" in syslog. dyld logs are pretty helpful at explaining why it couldn't load something.
